var/spack/repos/builtin/packages/soapdenovo2/package.py

```diff
@@ -15,16 +17,19 @@ class Soapdenovo2(MakefilePackage):
 
     homepage = "https://github.com/aquaskyline/SOAPdenovo2"
     url = "https://github.com/aquaskyline/SOAPdenovo2/archive/r240.tar.gz"
+    maintainers("snehring")
 
     version("242", sha256="a0043ceb41bc17a1c3fd2b8abe4f9029a60ad3edceb2b15af3c2cfabd36aa11b")
     version("240", sha256="cc9e9f216072c0bbcace5efdead947e1c3f41f09baec5508c7b90f933a090909")
 
     def flag_handler(self, name, flags):
-        if self.spec.satisfies("%gcc@10:"):
-            if name == "cflags" or name == "CFLAGS":
-                flags.append("-fcommon")
-            if name == "cxxflags" or name == "CXXFLAGS":
-                flags.append("-fcommon")
+        if name.lower() == "cflags" and self.spec.satisfies("%gcc@10:"):
+            flags.append("-fcommon")
+        if name.lower() == "cflags" or name.lower() == "cxxflags" or name.lower() == "cppflags":
+            opt_flag = re.compile("-O.*")
+            # This package cannot compile with anything other than the -O3
```

Comment:
Don't you mean the package cannot compile with any *optimization* flags/levels other than the `-O3` in the makefile?

Reply:
Yep. I've changed the wording a bit to be more clear.